src/turtle/turtle.ts

```diff
@@ -87,10 +87,19 @@ export class Turtle {
     this.direction = v.addAngles(this.direction, angle)
   }
 
-  optimize(iterations: number) {
-    const ev = k.optimize(this.loss, this.params, iterations)
-    this.params = ev.params
-  }
+  optimize: (iterations: number) => void = (() => {
```

Comment:
this took me a bit to parse. I'm ok with either, but maybe just put `currLoss` and `opt` in as inst vars of the class? I tend to shy away from classes in TS for most things but since we're using one already, might as well make use of its state.